src/engine/TPA.cc

```diff
@@ -1049,6 +1137,54 @@ bool TPABase::checkLessThanFixedPoint(unsigned short power) {
                 return true;
             }
         }
+        // now check the produced if transition invariants are actually safety invariants
+        {
+            MainSolver solver(logic, config, "Fixed-point checker");
+            solver.insertFormula(logic.mkAnd({init, logic.mkAnd(rightInvariants), getNextVersion(query)}));
+            auto satres = solver.check();
+            bool restrictedInvariant = false;
+            if (satres == s_False) {
+                if (verbose() > 0) {
+                    std::cout << "; Left fixed point detected in less-than relation on level " << i << " from " << power
+                              << std::endl;
+                    std::cout << "; Fixed point detected for "
+                              << (not restrictedInvariant ? "whole transition relation"
+                                                          : "transition relation restricted to bad")
+                              << std::endl;
+                }
+                explanation.invariantType = restrictedInvariant
+                                                ? SafetyExplanation::TransitionInvariantType::RESTRICTED_TO_QUERY
+                                                : SafetyExplanation::TransitionInvariantType::UNRESTRICTED;
+                explanation.relationType = TPAType::LESS_THAN;
+                explanation.power = i;
```

Comment:
This does not make sense anymore, because the invariants can come from different levels.